core/rootchain.go

```diff
@@ -0,0 +1,127 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	log "github.com/sirupsen/logrus"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	// "github.com/ethereum/go-ethereum/common/hexutil"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/icstglobal/go-icst/chain"
+	"github.com/icstglobal/go-icst/chain/eth"
+	"github.com/icstglobal/plasma/core/types"
+
+	"strings"
+	"time"
+)
+
+const (
+	DepositEventName     = "Deposit"
+	ExitedStartEventName = "ExistedStart"
+)
+
+type RootChain struct {
+	chain    chain.Chain
+	sub      map[string]func(eventName string, log ethtypes.Log) // map topic0 to name
+	cxAbi    abi.ABI
+	operator *Operator
+}
+
+type DepositEvent struct {
+	Depositor    common.Address
+	DepositBlock *big.Int
+	Token        common.Address
+	Amount       *big.Int
+	Raw          ethtypes.Log // Blockchain specific contextual infos
+}
+
+// chain
+func NewRootChain(url string, abiStr string, operator *Operator) (*RootChain, error) {
+
+	client, err := ethclient.Dial(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect eth rpc endpoint {%v}, err is:%v", url, err)
+	}
+	blc := eth.NewChainEthereum(client)
+	chain.Set(chain.Eth, blc)
+	// parse abi
+	abiParsed, err := abi.JSON(strings.NewReader(abiStr))
+	if err != nil {
+		return nil, err
+	}
+	rc := &RootChain{
+		chain:    blc,
+		sub:      make(map[string]func(eventName string, log ethtypes.Log)),
+		cxAbi:    abiParsed,
+		operator: operator,
+	}
+	// register dealing func
+	rc.sub[DepositEventName] = rc.dealWithDepositEvent
+	rc.sub[ExitedStartEventName] = rc.dealWithExitStartedEvent
+
+	// start loop to sync root chain event
+	go rc.loopEvent(DepositEventName)
+	go rc.loopEvent(ExitedStartEventName)
+
+	return rc, nil
+}
+
+func (rc *RootChain) loopEvent(eventName string) {
+	fromBlock := big.NewInt(100)
```

Comment:
why 100? it's too small, there will be huge number of logs to be processed, but not interest to the Plasma chain. It's better to config this value, and we can change it to the latest block num of Ethereum mainnet when we launch Plasma.

Reply:
I will achieve this in the next feature.